support-task/src/test/resources/supportProcess.bpmn

```diff
@@ -20,7 +20,7 @@
       <bpmn2:messageEventDefinition id="MessageEventDefinition_069i0vi" messageRef="Message_0zbmusz" />
     </bpmn2:boundaryEvent>
   </bpmn2:process>
-  <bpmn2:message id="Message_0zbmusz" name="cancelSupportTask" />
+  <bpmn2:message id="Message_0zbmusz" name="cancelSupportTask_${mainTaskInstanceId}_${supportTaskTypeKey}" />
```

Comment:
Why not use correlationKeys? I know it is typically used to correlate with process variables but it may also work with task variables. I'm not sure though.
If it works then it is a cleaner solution than to fiddle with message names.

Reply:
yes, it works - for correlation keys, business key and message name. The idea is the message name to be unique to contain actually the process variables - maintaskid and taskdef

Reply:
I correlated it finally with support process Id. I've tried to do it with  .localVariablesEqual(correlationKeys) but I've received MismatchingMessageCorrelationException. I didn't try all possible combinations to set localvariables.

Reply:
That is why I told you to create a separate module and test only the technical part you are focusing on instead of immediately integrating in the combined functional part.